lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,12 +185,321 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
+            )
+            unpacked_dir = server_config.INCOMING / "ABC"
+            with pytest.raises(TarballUnpackError) as exc:
+                tree.unpack(md5)
+            assert (
+                str(exc.value)
+                == f"An error occurred while unpacking {tarball_path}: Error moving '{unpacking_dir}' to '{unpacked_dir}': No such file or directory: '{unpacking_dir}'"
+            )
+
+    def test_unpack_move_dest(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Show that, when unpacked directory for moving results is not present and
+        raises FileNotFoundError Exception, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+
+        def mock_move(src: pathlib.Path, dest: pathlib.Path):
+            assert src == (
+                server_config.INCOMING / "ABC" / f"{tarball_name}.unpack" / tarball_name
+            )
+            raise FileNotFoundError(f"No such file or directory: '{dest}'")
+
+        def mock_run(args, **kwargs):
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Success!", stderr=None
+            )
+
+        with monkeypatch.context() as m:
+            m.setattr("subprocess.run", mock_run)
+            m.setattr(shutil, "move", mock_move)
+            m.setattr(shutil, "rmtree", lambda path, **kwargs: None)
+            tree = FileTree(server_config, make_logger)
+            tree.create("ABC", source_tarball)
+
+            tarball = tree.find_dataset(md5)
+            tarball_path = server_config.ARCHIVE / "ABC" / source_tarball.name
+            unpacking_dir = (
+                server_config.INCOMING / "ABC" / f"{tarball_name}.unpack" / tarball_name
+            )
+            unpacked_dir = server_config.INCOMING / "ABC"
+            with pytest.raises(TarballUnpackError) as exc:
+                tree.unpack(md5)
+            assert (
+                str(exc.value)
+                == f"An error occurred while unpacking {tarball_path}: Error moving '{unpacking_dir}' to '{unpacked_dir}': No such file or directory: '{unpacked_dir}'"
+            )
+
+    def test_unpack_move_nested_remove(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Show that, when we cannot remove partially unpacked directory and
+        raises FileNotFoundError Exception, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+
+        def mock_move(src: pathlib.Path, dest: pathlib.Path):
+            raise FileNotFoundError(f"No such file or directory: '{src}'")
+
+        def mock_rmtree(path: pathlib.Path, **kwargs):
+            if path == (server_config.INCOMING / "ABC" / tarball_name):
+                raise FileNotFoundError(f"No such file or directory: '{path}'")
```

Comment:
As things currently stand, `rmtree()` should be called _twice_ in this scenario -- once to remove `server_config.INCOMING / "ABC" / tarball_name` and once to remove `server_config.INCOMING / "ABC" / f"{tarball_name}.unpack" / tarball_name` (I _think_...).

I feel like this mock should only accept those two paths, and it should assert an error if any other path shows up here.  (And, unless you change the CUT so that we don't have `rmtree()` raising exceptions, you probably need two scenarios to test an exception raised by the first call vs. an exception raised by the second.)